AirSkyBoat/AirSkyBoat
# Fix minstrels ring spellcast reduction and hp latent check

**_I affirm:_**

## What does this pull request do?

Closes #1905 

* Changes minstrel ring latent value to 25 instead of -25; the mod expects positive values for spellcast reduction, and negative values causes an underflow to give full 50% cast reduction
* Changes latent checks to use existing GetHPP methods which calculate actual maxhp; the existing maxhp did not take into account gear whatsoever

## Steps to test these changes


1. Be taru brd75/rdm37
2. Equip enough gear to hit hp latent
3. Cast valor minuet, observe cast goes off at 6s (I also logged this with debug messages to ensure "cast" was set to 6s)

![image](https://user-images.githubusercontent.com/5553731/205027160-c0b54ee2-80d1-4dbd-a7bb-da7a60357805.png)

4. Equip convert gear, observe cast goes off at 8s (also logged with debug messages to ensure latent did not trigger)

![image](https://user-images.githubusercontent.com/5553731/205027530-e3c170ff-61d7-4793-bbdb-e5b3305277c1.png)
